dask/array/core.py

```diff
@@ -2585,15 +2645,41 @@ def store(x, out, index, lock, region):
             if lock:
                 lock.release()
 
-        return None
+        return result
 
     slices = slices_from_chunks(arr.chunks)
 
     name = 'store-%s' % arr.name
     dsk = dict()
     for t, slc in zip(core.flatten(arr.__dask_keys__()), slices):
         store_key = (name,) + t[1:]
-        dsk[store_key] = (store, t, out, slc, lock, region)
+        dsk[store_key] = (store, t, out, slc, lock, region, keep)
+
+    return dsk
+
+
+def retrieve_from_ooc(out_dsk):
```

Comment:
This could use a docstring

Reply:
That's fine. Again this is mirroring `insert_to_ooc`, which didn't have one either. Should `insert_to_ooc` have a docstring? FWIW this is part of the private API and shouldn't be user facing.

Reply:
It is handy when reviewing.  I had no idea what `out_dsk` represented for example.